### ISSUE TITLE
Can't install via Xposed Installer due to modified package name

### ISSUE THREAD

**Issue by blackwind** [phillipberndt/ChooseableOutgoingCalls#4]
Package name is incorrect (downloaded: com.pberndt.Choos**e**ableOutgoingCalls, expected: com.pberndt.ChoosableOutgoingCalls)

**Comment by phillipberndt** [comment]
Thanks for letting me know. I uploaded a new version with the typo reintroduced (the repository won't let me fix the package's name). Since I'm not using using Android 4 anymore, I can't test if the new version still works. Let me know if I accidentally broke anything. (Which is likely, because I switched to Android Studio)

**Comment by blackwind** [comment]
Ah, I found this through the Google Code bug and didn't read the description in Xposed until just now. Didn't realize Android 5 wasn't supported. Any chance you'll have time to take a closer look at that any time soon?

**Comment by phillipberndt** [comment]
I know how to fix this, probably: [Looking at the source code](https://android.googlesource.com/platform/packages/apps/ContactsCommon/+/master/src/com/android/contacts/common/CallUtil.java#68), it appears they changed the signature of `com.android.contacts.common.CallUtil.getCallIntent` from `Uri, String` to just `Uri`.

*But* I can't test this without much effor, because I don't currently use Xposed. I'll attach an experimental APK based on the latest code I've committed a minute ago. Could you give it a try and let me know if it works? If it does, selecting & calling a contact from the builtin contacts app (Which is `com.android.contacts`, not `com.google.android.contacts` - I don't know if the latter reuses the code, will check shortly) should pop up a selection of all apps that handle `tel:` URLs, the same you see when you click on a tel URL in your browser.

https://www.androidfilehost.com/?fid=24352994023703677

**Comment by blackwind** [comment]
No luck. Still jumps straight to the cellular dialer app.

**Comment by phillipberndt** [comment]
Which other app do you have installed that can handle tel: URLs with VIEW intents? (Skype can, for example, but Hangouts & Whatsapp can not. `am start tel:1234` from a shell should show you a list.)

**Comment by blackwind** [comment]
Hangouts *Dialer* shows up, as does TextNow and Phone.

**Comment by phillipberndt** [comment]
Great, thanks. I'll look into it as soon as I find the time to play with
xposed.

>

**Comment by blackwind** [comment]
Much appreciated. To my eyes, at least, it looks like what you've already come up with *should* work, so I'm very interested to see the end result.

**Comment by phillipberndt** [comment]
I've digged a bit deeper into this. The module didn't work because Google's variant of the package is actually called `com.google.android.contacts`. Also, the [Android L function looks different](https://android.googlesource.com/platform/packages/apps/ContactsCommon/+/lollipop-dev/src/com/android/contacts/common/CallUtil.java#123) from the M one from master.

Here's an updated version: https://www.androidfilehost.com/?fid=24352994023704270

Unfortunately, I still cannot test this, because I already updated to Marschmallow, which changed the signature again (but also doesn't suffer from this bug anymore).

**Comment by phillipberndt** [comment]
I've digged a bit deeper into this. There's some weird stuff going on. The module didn't work because Google's variant of the package is actually called `com.google.android.contacts`. Also, the [Android L function looks different](https://android.googlesource.com/platform/packages/apps/ContactsCommon/+/lollipop-dev/src/com/android/contacts/common/CallUtil.java#123) from the M one from master. The good news is that Marschmallow does not longer require this patch at all, because they finally replaced the CALL_PRIVILEGED intent with a simple CALL. Lollipop leaves me confused. The source (and also the decompiled contacts APK) references [a PhoneAccountHandle class](http://developer.android.com/reference/android/telecom/PhoneAccountHandle.html), but the documentation and Android Studio say that this API isn't available before Marschmallow.

My current workaround is to try to hook all getCallIntent() methods and return a modified version of the intent, with the action replaced by VIEW. I cannot test this, unfortunately, because I already upgraded to M.

Here's the updated test file: https://www.androidfilehost.com/?fid=24352994023704288
